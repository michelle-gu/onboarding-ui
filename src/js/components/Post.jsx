import React, { Component } from "react";
import ReactDOM from "react-dom";

class Post extends Component {

  	constructor() {
    	super();
  	}

    getCreatedAt() {
      let options = { month: 'short', day: 'numeric' };
      let createdAt  = new Date(this.props.createdAt);
      return createdAt.toLocaleDateString("en-US", options);
    }

  	render() {
    	return (
    		<div className="user">
    			<img id="profile-img" src={this.props.profileImgUrl}></img>
    			<span className="name">{this.props.name}</span>
    			<span className="name small-text">{this.props.twitterHandle}</span>
    		</div>

        <a>
      		<div className="tweet">
      			<div className="date small-text">{this.getCreatedAt()}</div>
      			<div>{this.props.message}</div>
      		</div>
        </a>
    	);
  	}

}

Post.defaultProps = {
	profileImgUrl: "",
  	name: "name",
  	twitterHandle: "twitterHandle",
};

export default Post;
