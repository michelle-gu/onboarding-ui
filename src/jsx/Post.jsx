import React, { Component } from "react";

const options = { month: 'short', day: 'numeric' };

class Post extends Component {

    constructor(props) {
      super(props);

      this.createdAt = new Date(this.props.createdAt).toLocaleDateString("en-US", options);
    }

    render() {
      return (
        <div className="post">
          <div className="user">
            <img id="profile-img" src={this.props.profileImageUrl}></img>
            <span className="name">{this.props.name}</span>
            <span className="name small-text">{this.props.twitterHandle}</span>
          </div>

          <a className="link" href={this.props.url} target="_blank">
            <div className="tweet">
              <div className="date small-text">{this.createdAt}</div>
              <div>{this.props.message}</div>
            </div>
          </a>
        </div>
      );
    }

}

export default Post;
