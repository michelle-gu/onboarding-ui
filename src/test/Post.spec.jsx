import React from 'react';
import { shallow } from 'enzyme';
import Post from '../jsx/Post';

describe('Post', () => {
    let wrapper;
    beforeEach(() => wrapper = shallow(<Post profileImageUrl={''} 
                                             name={''} 
                                             twitterHandle={''} 
                                             createdAt={0} 
                                             message={''} 
                                             url={''} />));

    it('should render correctly', () => expect(wrapper).toMatchSnapshot());

    it('should render 5 <div />', () => {
        expect(wrapper.find('div').length).toEqual(5);
    });

    it('should render 1 <img />', () => {
        expect(wrapper.find('img').length).toEqual(1);
    });

    it('should render 2 <span />', () => {
        expect(wrapper.find('span').length).toEqual(2);
    });

    it('should render 1 <a />', () => {
        expect(wrapper.find('a').length).toEqual(1);
    });

});
