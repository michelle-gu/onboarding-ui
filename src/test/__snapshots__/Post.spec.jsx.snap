// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render correctly 1`] = `
<div
  className="post"
>
  <div
    className="user"
  >
    <img
      id="profile-img"
      src=""
    />
    <span
      className="name"
    />
    <span
      className="name small-text"
    />
  </div>
  <a
    className="link"
    href=""
    target="_blank"
  >
    <div
      className="tweet"
    >
      <div
        className="date small-text"
      >
        Dec 31
      </div>
      <div />
    </div>
  </a>
</div>
`;
